include/tvm/relay/attrs/nn.h

```diff
@@ -173,6 +173,26 @@ struct UpSamplingAttrs : public tvm::AttrsNode<UpSamplingAttrs> {
 };
 
 
+/*! \brief Attributes for dense operator */
+struct DenseAttrs : public tvm::AttrsNode<DenseAttrs> {
+  int units;
```

Comment:
make units optional(None) instead of 0 as None so use IndexExpr here, see conv2d as an example